File: backend/Expense_tracker_service/java/com/monexel/expensetracker/service/IncomeService.java
package com.monexel.expensetracker.service;

import java.time.LocalDate;
import java.util.List;

import com.monexel.expensetracker.request.IncomeRequest;
import com.monexel.expensetracker.response.IncomeResponse;

public interface IncomeService {

	IncomeResponse addIncome(IncomeRequest request);
    IncomeResponse updateIncome(Long id, IncomeRequest request);
    void deleteIncome(Long id);
    IncomeResponse getIncomeById(Long id);
    List<IncomeResponse> getAllIncomesByUser(Long userId,LocalDate startDate, LocalDate endDate);

}

File: frontend/src/services/incomeService.js
import api from "./api";
 
export const addIncome = async (incomeData) => {
  const response = await api.post("/api/income/addIncome", incomeData);
  return response.data;
};
 
export const updateIncome = async (id, incomeData) => {
  const response = await api.put(`/api/income/updateIncome/${id}`, incomeData);
  return response.data;
};
 
export const deleteIncome = async (id) => {
  await api.delete(`/api/income/deleteIncome/${id}`);
};
 
export const getIncomeById = async (id) => {
  const response = await api.get(`/api/income/getIncomeById/${id}`);
  return response.data;
};
 
export const getAllIncomeByUser = async (userId,  startDate, endDate) => {
  const response = await api.get(`/api/income/getAllIncomeByUser/${userId}?startDate=${startDate}&endDate=${endDate}`);
  return response.data;
};